pytest_reportportal/plugin.py

```diff
@@ -55,8 +55,8 @@ def pytest_configure_node(node):
     if node.config._reportportal_configured is False:
         # Stop now if the plugin is not properly configured
         return
-    node.slaveinput['py_test_service'] = pickle.dumps(node.config.
-                                                      py_test_service)
+    node.workerinput['py_test_service'] = pickle.dumps(
```

Comment:
It will break compatibility with xdist versions below 2.0.0. Also, it's required to change the is_master function and the pytest_configure hook where we are doing pickle.loads.

Let's do something like below:

```suggestion
    workerinput = getattr(node, 'workerinput', node.slaveinput)
    workerinput['py_test_service'] = pickle.dumps(
```

Reply:
> It will break compatibility with xdist versions below 2.0.0 when the deprecated attribute slaveinput is removed.
it actually should not, the `slaveinput` and `workerinput` have coexisted in versions below 2.0.0 for a long time:
https://github.com/pytest-dev/pytest-xdist/pull/541/files#diff-8db1df3356c9e2971831603998e197e0015b6cb52edcc77eb45ef00210440768L228

Reply:
```
(rp_agent_xdist1)    rplevka@localhost  test_totok  $ pytest --reportportal -n3 --count=3 test_totok.py 
================================================= test session starts =================================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/rplevka/work/playground/test_totok, inifile: pytest.ini
plugins: forked-1.3.0, xdist-1.34.0, reportportal-5.0.6, repeat-0.9.1
gw0 [3] / gw1 [3] / gw2 [3]
...                                                                                                             [100%]
================================================== 3 passed in 0.54s ==================================================
```

Reply:
I addressed the rest of the comment - replaced the occurrences of `slaveinput` by `workerinput`.
it's still running strong on the xdist<2:
```
$ pytest --reportportal -n3 --count=3 
====== test session starts =====
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/rplevka/work/playground/test_totok, inifile: pytest.ini
plugins: forked-1.3.0, xdist-1.34.0, reportportal-5.0.6, repeat-0.9.1
gw0 [3] / gw1 [3] / gw2 [3]
...                                                                                                             [100%]
===== 3 passed in 0.58s =====
```

as well as the new pytest+pytest-xdist:

```
$ pytest --reportportal -n3 --count=3 
====== test session starts =====
platform linux -- Python 3.7.9, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/rplevka/work/playground/test_totok, configfile: pytest.ini
plugins: xdist-2.1.0, forked-1.3.0, reportportal-5.0.6, repeat-0.9.1
gw0 [3] / gw1 [3] / gw2 [3]
...                                                                                                             [100%]
===== 3 passed in 0.65s =====
```
pytest.ini i used:
```
$ cat pytest.ini 
[pytest]
rp_endpoint = http://localhost:8080
rp_uuid = 5bb7af29-e5a4-4400-bbd5-936253db720c
rp_project = superadmin_personal

rp_launch = old_xdist

rp_hierarchy_module = True
```

Reply:
the xdist is properly recognized and the hierarchy disabled:

![rp](https://i.imgur.com/2Q4yb7z.png)